src/main/java/org/springframework/hateoas/UriTemplate.java

```diff
@@ -67,6 +67,7 @@ public UriTemplate(String template) {
 			String[] names = matcher.group(2).split(",");
 
 			for (String name : names) {
+				name = name.endsWith("*")? name.substring(0, name.length() - 1) : name;
```

Comment:
I'm not eager to accept a change in the UriTemplate pattern that also includes a special rule outside the regex to enforce it. I'd also have to read the UriTemplate spec to ensure this fits the spec as well.

To my understanding, UriTemplate is a complex spec with multiple levels, hence getting into it needs to have a well defined reason. That's why I am slow to respond to this one.